request.js

```diff
@@ -383,8 +377,24 @@ Request.prototype.init = function (options) {
     self.proxy = getProxyFromURI(self.uri)
   }
 
-  // Pass in `tunnel:true` to *always* tunnel through proxies
-  self.tunnel = !!options.tunnel
+  // Goals:
+  //  - Tunnel HTTPS by default, or if a previous request in the redirect chain
+  //    was tunneled
+  //  - Allow the user to override this setting
+  // Implementation:
+  //  - If this is a redirect, self.tunnel may be set -> don't do anything.
+  //  - If options.tunnel is set, use it.
+  //  - Otherwise, tunnel if the destination is HTTPS, but don't set
+  //    tunnel=false if it's not, because if a later rquest in the redirect
+  //    chain is HTTPS then that request (and any subsequent ones) should be
+  //    tunneled.
+  if (typeof self.tunnel === 'undefined') {
```

Comment:
This nested if statement is the only thing in the code that's catching my eye.
It's not bad by any means, and pretty much necessary given the way we're operating on `self.tunnel`.

However I just wanted to toss this idea in.
What if we used a function that took in self and options, then returned a boolean.
Then in the code we would call that function and assign its value to self.tunnel.

Like this more or less:
```javascript
function setTunnel(self, options) {
  if(self.tunnel) {
    return self.tunnel
  }

  if(options.tunnel) {
    return options.tunnel
  }

  if(self.uri.protocol === 'https:') {
    return true
  }

  return false // not necessary since undefined is falsy
}

// else where in code

self.tunnel = setTunnel(self, options)
```

What do you think?